### PR TITLE
remove inline pylint ignore statements

### PR DESCRIPTION
**Pull request by GiancarloFusiello**
Removes inline pylint ignore statements and disables some pylint checks that are covered by MyPy and Flake8.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -9,8 +9,9 @@ sections = FUTURE,STDLIB,FLASK,THIRDPARTY,FIRSTPARTY,LOCALFOLDER
 not_skip = __init__.py
 
 [pylint]
-disable = missing-docstring
+disable = missing-docstring,no-member,redefined-builtin,redefined-outer-name,unused-variable
```
Comment (giorgiosironi) [inline setup.cfg:12]: `no-member` would be quite a good check, but if the libraries don't support it well it quickly becomes incompatible with using them; I remember even `requests` having problems on the `Response` object.

Comment (giorgiosironi) [reply setup.cfg:12]: I still don't understand where `unused-variable` came from, trying it out locally.

Comment (giorgiosironi) [reply setup.cfg:12]: For posterity:
```
************* Module search.ping
search/ping.py:8:4: W0612: Unused variable 'ping' (unused-variable)
************* Module search.errors
search/errors.py:28:4: W0612: Unused variable 'error' (unused-variable)
************* Module search.search
search/search.py:13:4: W0612: Unused variable 'search' (unused-variable)
```
These are all functions defined inside another function, so they are considered a local variable that is not called. Flask uses the `@blueprint.route` annotation to get a reference to them, and that's how they escape from the blueprint function.

**Review by giorgiosironi — APPROVED** [review]

File: setup.cfg (hunk 1 above)
Comment (GiancarloFusiello) [reply setup.cfg:12]: I have checked each of the rules and, given the context I would expect, MyPy and Flake8 do flag the errors correctly.

**Review by thewilkybarkid — APPROVED** [review]

**Pull request merged**